ovis: example: aruco_ar_demo.cpp references out-dated header.

- OpenCV => 4.x
- Operating System / Platform => Ubuntu 24.04
- Compiler => GCC 13.2

```
-- General configuration for OpenCV 4.9.0-dev =====================================
--   Version control:               4.9.0-415-gcbf71c380f
--
--   Extra modules:
--     Location (extra):            /home/kmtr/work/opencv_contrib4/modules
--     Version control (extra):     4.9.0-68-g3c2bcbfe
```

##### Detailed description

This sample program is out of date to suitable currently libraries.

```
/home/kmtr/work/opencv_contrib4/modules/ovis/samples/aruco_ar_demo.cpp:6:10: fatal error: opencv2/aruco_detector.hpp: No such file or directory
    6 | #include <opencv2/aruco_detector.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

Until it depends, skip it.

See https://github.com/opencv/opencv/issues/23329

##### Steps to reproduce

```
sudo apt install libogre-1.12-dev
cmake -B build4-full2 -S opencv4 -DOPENCV_EXTRA_MODULES_PATH=opencv_contrib4/modules -DBUILD_EXAMPLES=ON
cmake --build 
```

##### Issue submission checklist

 - [x] I report the issue, it's not a question
 - [x] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found any solution
 - [x] I updated to the latest OpenCV version and the issue is still there
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

--- a/modules/ovis/CMakeLists.txt
+++ b/modules/ovis/CMakeLists.txt
@@ -24,7 +24,7 @@ ocv_glob_module_sources()
 ocv_module_include_directories()
 ocv_create_module()
 
-ocv_add_samples(opencv_aruco)
+ocv_add_samples(opencv_objdetect opencv_aruco)
 
 ocv_warnings_disable(CMAKE_CXX_FLAGS -Wunused-parameter)
 ocv_target_link_libraries(${the_module} ${OGRE_LIBRARIES})

--- a/modules/ovis/samples/aruco_ar_demo.cpp
+++ b/modules/ovis/samples/aruco_ar_demo.cpp
@@ -3,7 +3,7 @@
 #include <opencv2/videoio.hpp>
 
 #include <opencv2/ovis.hpp>
-#include <opencv2/aruco_detector.hpp>
+#include <opencv2/aruco.hpp>
 
 #include <iostream>
 
@@ -24,10 +24,12 @@ int main()
   const double focal_length = 800.0;
 
   // aruco
-  Ptr<aruco::Dictionary> adict = aruco::getPredefinedDictionary(aruco::DICT_4X4_50);
-  //Mat out_img;
-  //aruco::drawMarker(adict, 0, 400, out_img);
-  //imshow("marker", out_img);
+  aruco::Dictionary adict = aruco::getPredefinedDictionary(aruco::DICT_4X4_50);
+
+  aruco::ArucoDetector detector(adict);
+  Mat out_img;
+  adict.generateImageMarker(0, 400, out_img);
+  imshow("marker", out_img);
 
   // random calibration data, your mileage may vary
   Mat1d cm = Mat1d::zeros(3, 3); // init empty matrix
@@ -53,7 +55,7 @@ int main()
   while (ovis::waitKey(1) != KEY_ESCAPE) {
     cap.read(img);
     win->setBackground(img);
-    aruco::detectMarkers(img, adict, corners, ids);
+    detector.detectMarkers(img, corners, ids);
 
     waitKey(1);
 

--- a/modules/ovis/samples/aruco_ar_demo.py
+++ b/modules/ovis/samples/aruco_ar_demo.py
@@ -3,7 +3,7 @@ import cv2 as cv
 
 # aruco
 adict = cv.aruco.getPredefinedDictionary(cv.aruco.DICT_4X4_50)
-cv.imshow("marker", cv.aruco.drawMarker(adict, 0, 400))
+cv.imshow("marker", adict.generateImageMarker(0, 400))
 
 # random calibration data. your mileage may vary.
 imsize = (800, 600)

--- a/modules/ovis/src/ovis.cpp
+++ b/modules/ovis/src/ovis.cpp
@@ -588,6 +588,7 @@ public:
         cam->setDebugDisplayEnabled(true);
         cam->setNearClipDistance(1e-9);
         cam->setFarClipDistance(zFar);
+        cam->getFrustumExtents(); // force update
 
         _setCameraIntrinsics(cam, K, imsize);
 

